### PR TITLE
[PREVIEW] Read envelopes by status

### PR DESCRIPTION
**Pull request by rkondratowicz**
https://tools.hmcts.net/jira/browse/RPE-611

Will look into refactoring existing tests if being able to retrieve all envelopes (or envelopes in given status) will make tests more straigtforward.

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeControllerTest.java [hunk 1]
```diff
@@ -156,7 +157,7 @@ public void should_successfully_return_all_envelopes_with_processed_status_for_a
 
         given(tokenValidator.getServiceName("testServiceAuthHeader")).willReturn("test_service");
 
-        mockMvc.perform(get("/envelopes")
+        mockMvc.perform(get("/envelopes?status=" + Status.PROCESSED)
```
Comment (doncem) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeControllerTest.java]: we have a static import in this class for that status 😉

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeRetrieverTest.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static uk.gov.hmcts.reform.bulkscanprocessor.helper.EnvelopeCreator.envelope;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class EnvelopeRetrieverTest {
+
+    @Autowired private EnvelopeRepository envelopeRepo;
+    @Mock EnvelopeAccessProperties accessProperties;
```
Comment (doncem) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeRetrieverTest.java]: can we not mock the properties - it's 100% configurable

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/Envelope.java [hunk 3]
```diff
@@ -153,6 +153,10 @@ public String getZipFileName() {
         return zipFileName;
     }
 
+    public void setZipFileName(String zipFileName) {
+        this.zipFileName = zipFileName;
+    }
+
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/Envelope.java]: ha! just used that locally too 😺

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeRetrieverService.java [hunk 4]
```diff
@@ -31,14 +30,14 @@ public EnvelopeRetrieverService(
         this.envelopeAccessProperties = envelopeAccessProperties;
     }
 
-    public List<Envelope> getProcessedEnvelopesByJurisdiction(final String serviceName) {
+    public List<Envelope> findByServiceAndStatus(String serviceName, Status status) {
         log.info("Fetch requested for envelopes for service {}", serviceName);
 
         String jurisdiction = getJurisdictionByServiceName(serviceName);
 
-        log.info("Fetching all processed envelopes for service {} and jurisdiction {}", serviceName, jurisdiction);
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeRetrieverService.java]: 👏

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeRetrieverTest.java (hunk 2 above)
Comment (luigibk) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeRetrieverTest.java]: I personally prefer the mock solution in this case. It's easier to read. Also functional tests do the job without mocks already. Moreover removing the mocks in this case seems to me would only be testing that Spring is capable of loading the properties from the config file, which is really outside the scope of the test.

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeRetrieverServiceTest.java [hunk 5]
```diff
@@ -38,7 +41,7 @@ public void setUp() {
     }
 
     @Test
-    public void should_return_all_envelopes_successfully_for_a_given_jurisdiction() throws Exception {
+    public void should_return_all_envelopes_successfully_for_a_given_jurisdiction_and_status() throws Exception {
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeRetrieverServiceTest.java]: We could few one test case i.e when invalid status is passed

Comment (rkondratowicz) [reply src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeRetrieverServiceTest.java]: add 2 more tests here:
https://github.com/hmcts/bulk-scan-processor/pull/101/commits/e038d4ce70b7ac37c5c8a9119c6eabafe39f3ac7

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeRetrieverService.java (hunk 4 above)
Comment (luigibk) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeRetrieverService.java]: Shouldn't the log statement include the status?

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeRetrieverService.java]: fixed

**Pull request merged**